app/controllers/users/sessions_controller.rb

```diff
@@ -3,11 +3,16 @@ class SessionsController < Devise::SessionsController
     include ::ActionView::Helpers::DateHelper
 
     skip_before_action :session_expires_at, only: [:active]
+    skip_before_action :require_no_authentication, only: [:new]
     before_action :confirm_two_factor_authenticated, only: [:update]
 
     def new
-      analytics.track_event(Analytics::SIGN_IN_PAGE_VISIT)
-      super
+      if user_fully_authenticated?
+        return redirect_to after_sign_in_path_for(current_user) unless user_fully_authenticated?
```

Comment:
Ah the mistake was not deleting the if statement. Code Climate complains without the unless!